library/core/src/ops/mod.rs

```diff
@@ -17,9 +17,7 @@
 //! should have some resemblance to multiplication (and share expected
 //! properties like associativity).
 //!
-//! Note that the `&&` and `||` operators short-circuit, i.e., they only
-//! evaluate their second operand if it contributes to the result. Since this
-//! behavior is not enforceable by traits, `&&` and `||` are not supported as
+//! Note that the `&&` and `||` operators are currently not supported as
 //! overloadable operators.
```

Comment:
Since the obvious question on reading this is *why*, would it be worth a quick note about how it can't be done *with a trait having the same structure as `BitAnd`/`BitOr`*, or something?

1st draft, please improve or rewrite as you see fit: 
> (The trait structure used by `BitAnd` and `BitOr` eagerly evaluates both arguments to the operator, which is unacceptable for the short-circuiting logical operators, thus `&&` and `||` need to be done differently.)

Reply:
Your note doesn't answer "why" really either :). If you add something like "their design is still under their discussion", but care must be taken to not imply that it's already consensus that such traits should be added *at all*.

Reply:
That's fair.  I suppose it's more answering "why didn't you just copy-paste `BitAnd` since you have `&` overridable", which is different from "why haven't you made it work yet".

Reply:
Thanks for the update!